src/calendaragendamodel.cpp

```diff
@@ -96,9 +99,6 @@ void CalendarAgendaModel::setEndDate(const QDate &endDate)
 
 void CalendarAgendaModel::refresh()
 {
-    if (!mIsComplete)
-        return;
```

Comment:
This doesn't look right. refresh() is called from property setters which can happen before the component is complete, but those should not cause any handling in the intermediate state until all the properties are set.